### ISSUE TITLE
MySQL8: Queries in QR&R never take effect (due to integer display lengths deprecated)

### ISSUE THREAD

**Issue by pgorod** [salesagility/SuiteCRM#9568]
#### Issue
MySQL is deprecating using any sort of length for all kinds of integer fields (shortint, integer, etc). It just changes all such fields to `int` without any specified length.

This causes the QR&R to always detect an issue with lengths if the vardefs specifies them. Just an example of one field:
``` 
/* Table : acl_actions */
/*COLUMNS*/
/*MISMATCH WITH DATABASE - aclaccess -  ROW [name] => 'aclaccess'  [type] => 'int'  [len] => ''  */
/* VARDEF - aclaccess -  ROW[name] => 'aclaccess'  [vname] => 'LBL_ACCESS'  [type] => 'int'  [len] => '3'  [reportable] => '1'  [comment] => 'Number specifying access priority; highest access "wins"'  [dbType] => 'int'  [required] => ''  */
/* INDEXES */
ALTER TABLE acl_actions   modify COLUMN `aclaccess` int(3)  NULL ;
```

Running the query directly in phpMyAdmin causes this warning to be issued:

`Warning: #1681 Integer display width is deprecated and will be removed in a future release.` 

Even after the UPDATE query, the field definition will **not** be modified to `int(3)`, it will stay as just `int`.

#### Expected Behavior
QR&R "Execute query" should only be required once.

#### Actual Behavior
No matter how often the query is run, the database stays in a state that will provoke the same query to be proposed again and again.

#### Possible Fix
Somewhere in DBManager making the code ignore this case?

#### Steps to Reproduce
1. Upgrade a previously existing instance to MySQL v8
2. Go to QR & R
3. Execute Query to solve vardef mismatches - note the presence of many queries just fixing lengths of integers (all kinds of integers)
4. Go to the QR & R again and the query is re-proposed...

#### Your Environment
* SuiteCRM Version used: 7.12.5
* Environment name and version: PHP 7.3, MySQL 8.0
* Operating System and version: Ubuntu 20.04

**Comment by M4tDR** [comment]
Please, same issue here. (Same version for SuiteCRM and MySQL).
The official documentation say MySQL 8 is supported, but not completeley as I understand

**Comment by pgorod** [comment]
@M4tDR you are right. But note that this can safely be ignored; SuiteCRM works just fine, despite of this error. It's like a "false positive", it brings some confusion and doubt, but not actual problems.

**Comment by M4tDR** [comment]
Thanks @pgorod, I know is only a warning, but we planned a release upgrade to a customer that asked for it and I'm not happy to use MySQL 8 cause also customer may not likes all that warning, so at the moment we upgrade SuiteCRM at version 7.12.5 but still on mysql 5.7.

I think that if we really want to push our customer for upgrade we have to fix that.

Bye

**Comment by johnM2401** [comment]
Hey!

Thank you for getting in touch and raising this!

As this was resolved with the merged PR #9770 linked to this, I am going to close this off for now.

Please let us know if you are still seeing this, or any other issues, in the CRM on the latest releases.

Thank you for your contributions to the project!
